### ISSUE TITLE
Running any command produces no output

### ISSUE THREAD

**Issue by spoco2** [arzzen/git-quick-stats#115]
**Describe the bug**
Running git-quick-stats brings up the menu, but selecting any option produces only the heading and a 'total:' or similar final line, with no actual data. Except option 8 (All branches), which does produce output.

**To Reproduce**
1. run git-quick-stats
2. select any option other than 8
3. Notice that there is a title in red, but no content

**Expected behavior**
Application produces output

**Screenshots**
![image](https://user-images.githubusercontent.com/5827872/91099735-3452f400-e6a7-11ea-9a26-7a9d3347949c.png)

**Desktop (please complete the following information):**
 - OS: MacOS 10.15.6
 - Terminal: iTerm2
 - Using Powerline fonts for terminal
 - Have also tried in default terminal

**Additional context**

**Comment by tomice** [comment]
Please try the following:
```
git clone https://github.com/arzzen/git-quick-stats.git
cd git-quick-stats
./git-quick-stats

# Enter any menu option
```

Does it produce any stats, or are you still seeing no results?

**Comment by spoco2** [comment]
Still no results I'm afraid: 
![image](https://user-images.githubusercontent.com/5827872/91100943-6bc2a000-e6a9-11ea-8ac4-c6aca760a748.png)

**Comment by arzzen** [comment]
What Unix shell do you use (sh, bash, zsh, ash ...)? (`bash --version`, `zsh --version` etc)
What git version do you have installed? (`git --version`)

Make sure you have these tools installed 
`awk ; basename ; cat ; column ; echo ; git ; grep ; head ; seq ; sort ; tput ; tr ; uniq ; wc` and `bsdmainutils`.

Please try these git commands:
`git -c log.showSignature=false log --use-mailmap --pretty=%aN | head -n 100 | sort | uniq -c | sort -nr` 
`git  log --use-mailmap | grep -i Author: | cut -c9- `
`git for-each-ref --sort=committerdate refs/heads/ --format='[%(authordate:relative)] %(authorname) %(refname:short)' | cat -n`

**Comment by spoco2** [comment]
zsh 5.7.1 (x86_64-apple-darwin19.0)
git version 2.24.0

All of those commands work fine. At least seem to.

For the tools, using 'which' on them had them all appear, except grep is aliased to this, which I'm guessing is fine, but putting here incase it shows an issue for you `grep: aliased to grep  --color=auto --exclude-dir={.bzr,CVS,.git,.hg,.svn}`

And bsdmainutils is not installed

**Comment by tomice** [comment]
I am unable to replicate this issue. Tested on the following OSes and setting zsh as my main shell:

Ubuntu 20.04 LTS
Fedora 31
macOS 10.14.6 (tested in default Terminal app)

```sh
MacBook-Pro:git-quick-stats tom$ zsh --version
zsh 5.3 (x86_64-apple-darwin18.0)
MacBook-Pro:git-quick-stats tom$ git --version
git version 2.20.1 (Apple Git-117)
MacBook-Pro:git-quick-stats tom$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
Copyright (C) 2007 Free Software Foundation, Inc.
```

Does using it non-interactively work? Example usage: `git-quick-stats -T`

**Comment by spoco2** [comment]
Non interactive doesn't work either I'm afraid.

Thanks for looking. I'll have to put it down to some specific thing with my setup here!

**Comment by tomice** [comment]
Would it be possible to try the following?
1) [Create a new user account in macOS](https://support.apple.com/guide/mac-help/set-up-other-users-on-your-mac-mtusr001/mac). Open the terminal and follow the above steps in https://github.com/arzzen/git-quick-stats/issues/115#issuecomment-679388558

2) Try a [previous version](https://github.com/arzzen/git-quick-stats/releases) to see if any of those possibly work. If one does, please let us know as it'll help pinpoint what's going on.

This can help narrow down if it's an issue in macOS 10.15.6.

**Comment by spoco2** [comment]
Hey. I created a new account and cloned git quick stats and ran it and same issue. All within the standard mac Terminal.

I can't spend any more time on this today, but if I remember later I'll come back and try a few of the older versions to see if it works in one of them.

**Comment by terra9** [comment]
same here! It is not showing any output.

**Comment by tomice** [comment]
> same here! It is not showing any output.

Hello, can you provide OS specs, what terminal you are using, and what shell you are using? Thanks.

**Comment by terra9** [comment]
OS :: Ubuntu 20.04
Terminal :: Gnome Terminal
Shell :: zsh
All latest versions available.

**Comment by tomice** [comment]
Looks like it could be zsh-related. Could you try this version and see if it works for you?

https://github.com/arzzen/git-quick-stats/releases/tag/2.1.3

**Comment by terra9** [comment]
Still same issue...

**Comment by terra9** [comment]
tested on bash too...

**Comment by tomice** [comment]
I have Ubuntu 20.04 LTS and am able to perform this without any issue, so I am not entirely sure what is going on. If you have time, could you possibly try some older versions and see if any of them work? https://github.com/arzzen/git-quick-stats/tags
![image](https://user-images.githubusercontent.com/8818630/93275235-63cabd00-f78a-11ea-9188-115f617393b2.png)

**Comment by faried** [comment]
The problem is somewhere with the `--since` flag.  I haven't looked into why this happens but this change fixes it for me:
```diff
--- /usr/local/Cellar/git-quick-stats/2.1.4/bin/git-quick-stats 2020-07-22 20:18:59.000000000 +0500
+++ /tmp/git-quick-stats        2020-09-21 02:45:10.000000000 +0500
@@ -267,7 +267,7 @@
 
     git -c log.showSignature=false log ${_branch} --use-mailmap $_merges --numstat \
         --pretty="format:commit %H%nAuthor: %aN <%aE>%nDate:   %ad%n%n%w(0,4,4)%B%n" \
-        "$_since" "$_until" $_log_options $_pathspec | LC_ALL=C awk '
+        "$_until" $_log_options $_pathspec | LC_ALL=C awk '
         function printStats(author) {
         printf "\t%s:\n", author
```

(Yes, I know it's not the right fix, but I'm in the middle of filling out a self-review form for work, and that's why I used `git-quick-stats` today.)

Trying `env _GIT_SINCE='Thu Jan  1 00:00:00 1970' git-quick-stats` didn't work for me, either.

* MacOS 10.14.6
* Git 2.28.0 (from homebrew)
* bash 5.0.18(1)-release (from homebrew)

**Comment by arzzen** [comment]
@faried thank you for investigation
Maybe the main problem is in the default value... What do you think @tomice ? (line 17) https://github.com/arzzen/git-quick-stats/blob/8cea653d6cf7ec3bf4047d5efac1533598e37d8d/git-quick-stats#L10-L18

**Comment by tomice** [comment]
Hm, it's possible! Would anyone who is experiencing this problem be willing to test some range of date formats that work? The first git commit ever made was April 7th, 2005, so that could be a possible valid date. It could just be how date is formatted, too: https://github.com/git/git/commit/e83c5163316f89bfbde7d9ab23ca2e25604af290

Examples of date formats can be seen here for the GNU version:
https://man7.org/linux/man-pages/man1/date.1.html

And here it is for BSD:
https://www.freebsd.org/cgi/man.cgi?date

**Comment by yazver** [comment]
I have the same problem. 

```sh
git-quick-stats
```

and 

```sh
env _GIT_SINCE='Thu Jan 1 00:00:00 1970' git-quick-stats
```

nothing output.

This show info:
```sh
env _GIT_SINCE='Thu Jan 1 00:00:00 2020' git-quick-stats
```

**Comment by wollsi** [comment]
same here.

Fun story: It worked some time and stopped working. I did not change anything ... :/

**Comment by tomice** [comment]
Can someone who is experiencing this issue test this for me and see if it fixes it? I still cannot replicate this, unfortunately.

Attempted fix was applied to the master branch here:
https://github.com/tomice/git-quick-stats

**Comment by wollsi** [comment]
I reinstalled it and it started working again. I have no idea what happend. I will keep an eye on this. If it stops working again, I will test the fix.

**Comment by mukel** [comment]
Can confirm @tomice patch fixes it (Manjaro 20.2.1 x86_64 tested with zsh). It happens only for some repos, sadly, the only reproducer I have is a company private repo.
